### PR TITLE
remove deprecated dealii::Thread classes and functions

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

**Comment by tjhei** [comment]
```
/jenkins/workspace/dealii_PR-10315/source/base/thread_management.cc:155:3: error: ‘PosixThreadBarrier’ does not name a type
```

**Comment by tamiko** [comment]
So. This should be good to go now!

**Comment by drwells** [comment]
```
##[error]Response status code does not indicate success: 500 (Internal Server Error).
```
GitHub's CI is failing for unrelated reasons.

**Comment by masterleinad** [comment]
Yes, `GitHub` has issues. https://www.githubstatus.com/

File: include/deal.II/base/thread_management.h [hunk 1]
```diff
@@ -64,55 +64,21 @@ DEAL_II_NAMESPACE_OPEN
...
-   * used as a member variable to lock the other member variables of a
-   * class: in that case, the mutex of the copied-to object should
-   * only guard the members of the copied-to object, not the members
-   * of both the copied-to and copied-from object. Since at the time
-   * when the class is copied, the destination's member variable is
-   * not used yet, its corresponding mutex should also remain in its
-   * original state.
-   *
-   * @author Wolfgang Bangerth, 2002, 2003, 2009
+   * This class is a thin wrapper around `std::mutex`. The only difference
```
Comment (bangerth) [inline include/deal.II/base/thread_management.h:67]: Why replace the more descriptive documentation?

Comment (tamiko) [reply include/deal.II/base/thread_management.h:67]: I have kept almost all of the descriptive documentation in place and only removed the parts that will be factually incorrect after this pull request.

Comment (tamiko) [reply include/deal.II/base/thread_management.h:67]: If you want, I can put the `@author` clause back in place. I did not mean to offend you.

**Comment by tamiko** [comment]
I have to go through the testsuite. There are a number of tests that use deprecated functionality.

File: include/deal.II/base/thread_management.h (hunk 1 above)
Comment (bangerth) [reply include/deal.II/base/thread_management.h:67]: No, I don't actually care about the `@author` tag at all :-) I cared about the link to the wikipedia page, which I think gives useful background information...

Comment (tamiko) [reply include/deal.II/base/thread_management.h:67]: Ah! Let me put it back.

**Comment by tamiko** [comment]
@bangerth Mind if I hijack your pull request and rebase on top?

File: doc/news/changes/incompatibilities/20200522Maier [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Removed: The deprecated classes Threads::Mutex::ScopedLock,
+Threads::ConditionVariable, Threads::PosixThreadBarrier, and deprecated
+functions Threads::Mutex::acquire(), Threads::Mutex::release(),
+Threads::n_existing_threads(), Threads::this_thread_id() have been removed.
```
Comment (bangerth) [inline doc/news/changes/incompatibilities/20200522Maier:4]: I think you'll want to update that.

Comment (tamiko) [reply doc/news/changes/incompatibilities/20200522Maier:4]: Yes. I lost the fixup in a rebase.

**Comment by tamiko** [comment]
This now compiles on Windows.

**Pull request merged**